### PR TITLE
Allow linking SWIG Python for different platforms

### PR DESCRIPTION
**Pull request by dmikhalin**
## What is the goal of this PR?

We updated some dependencies to correspond with a newer `rules-rust` version, added a linux-arm64 platform, and fixed linking errors on Windows and errors with incorrect `libpython` dependency in the generated dynamic library.

### CODE REVIEW

File: builder/swig/python.bzl [hunk 1]
```diff
@@ -24,7 +24,8 @@ def _copy_to_bin(ctx, src, dst):
 
 def _swig_python_wrapper_impl(ctx):
     module_name = getattr(ctx.attr, "class_name", ctx.attr.name)
```
Comment (flyingsilverfin) [inline builder/swig/python.bzl]: Probably should be `module_name`

File: builder/swig/python.bzl [hunk 2]
```diff
@@ -133,28 +148,63 @@ def swig_python(name, lib, shared_lib_name=None, **kwargs):
...
+            mandatory = True,
+        ),
+        "src": attr.label(
+            mandatory = True,
+        ),
+    }
+)
+
+
+def py_native_lib_rename(name, out, src, visibility, **kwargs):
```
Comment (flyingsilverfin) [inline builder/swig/python.bzl:196]: Let's put a comment in the code here to save ourselves lots of pain in the future

File: builder/swig/python.bzl [hunk 3]
```diff
@@ -123,38 +134,73 @@ swig_python_wrapper = rule(
...
-        actual = select({
-            "@vaticle_dependencies//util/platform:is_mac": (shared_lib_name + ".so"),
-            "@vaticle_dependencies//util/platform:is_linux": (shared_lib_name + ".so"),
-            "@vaticle_dependencies//util/platform:is_windows": (shared_lib_name + ".lib"),
-        })
+        deps = [lib, swig_wrapper_name],
+        srcs = [swig_wrapper_name],
+        linkshared = True,
+        linkopts = select({
+            "@vaticle_dependencies//util/platform:is_windows": ["ntdll.lib"],
```
Comment (flyingsilverfin) [inline builder/swig/python.bzl:162]: Let's leave a comment here in the code

**Review by flyingsilverfin — APPROVED** [review]

**Comment by flyingsilverfin** [comment]
Title / goal should mention SWIG and Python to provide context

File: builder/swig/python.bzl (hunk 3 above)
Comment (dmikhalin) [reply builder/swig/python.bzl:162]: `ntdll.lib` is necessary when linking in Windows, so we need to specify this `linkopts` argument for the linker.

File: builder/swig/python.bzl (hunk 2 above)
Comment (dmikhalin) [reply builder/swig/python.bzl:196]: The generated dynamic library has to be copied into the output directory in order to be accessible by other rules.
We also choose the correct extension for the file.

File: builder/swig/python.bzl [hunk 4]
```diff
@@ -123,38 +134,73 @@ swig_python_wrapper = rule(
...
+            "//conditions:default": [],
+        }),
+        copts = select({
+            "@vaticle_dependencies//util/platform:is_mac": ["-undefined", "dynamic_lookup"],
```
Comment (dmikhalin) [inline builder/swig/python.bzl:166]: We need to allow weak import when linking on MacOS.

**Review by flyingsilverfin — APPROVED** [review]

**Pull request merged**